### ISSUE TITLE
Truffle not working with new Geth/v1.4.3-stable

### ISSUE THREAD

**Issue by msbachler** [ConsenSys/truffle#156]
I ran an apt-get update this morning on one of my blockchain servers and it upgraded me to Geth  v1.4.3-stable. I am on Truffle v 1.0.3. When I rebuilt one of my dapps and then ran it in the browser I got the following error:

app.js:5318 Uncaught TypeError: Cannot read property 'filter' of undefined

To help you locate it the actual code snippet in app.js around that error is:

,u=function(t){t.abi.filter(function(t){return"function"===t.type}).map(function(e){return new a(t._eth,e,t.address)}).forEach(function(e){e.attachToContract(t)})},

Just to be absolutely sure I rebuild a different dapp as well and ran that and got exactly the same error message as above. 

I have the same two dapps on a different server still running running Geth 1.3.6 with Truffle v1.0.3 and they run fine and don't get that error.

**Comment by msbachler** [comment]
I put some console logging to your app.js generated code. It is specifically upset about the object 't' in the above code snippet because it does not have the property 'abi' on which it wants to get the 'filter'. I printed out 't' and it is:

t:{"_eth":{"_requestManager":{"provider":{"host":"http://212.219.130.124:54200"},"polls":{},"timeout":null},"compile":{}},"transactionHash":null,"address":"0xd1bb9301e1a2d24965ef8ce1de7f9353946d5621"}

No idea if this helps you. I will keep digging. This is a major issue for me so I really need to get it fixed. I particularly want to know if there is anything I can do by just changing my code in some way to fix this.

**Comment by msbachler** [comment]
OK. It is failing at the point of trying to create a new instance of a contract. I put some console logs into the following code from your app.js:

	Pudding.at = function(address) {
	  var web3 = Pudding.getWeb3();
	  var contract_class = web3.eth.contract(this.prototype.abi);
	  var contract = contract_class.at(address);
	  return new this(contract);
	};

'this.prototype' is an empty object, so does not have an abi property when calling web3.eth.contract. Both the contract js code snippets for my two contracts have definitely been copied into the bottom of app.js as usual. I don't know enough about Pudding or your app.js really to know what is failing exactly. I will keep digging.

**Comment by msbachler** [comment]
I compared the app.js on my new blockchain server that works with the one on the old blockchain server that has been upgraded. Both return '1.0.3' for the 'truffle version' command. But the <contract>.sol.js code is quite different on both machines. Person.console.js on the new server says in the contract_data object that it was 'generated_with:"2.0.6"'. On the old server it says 'Person.generated_with = "1.0.3"'? But both say they are truffle version 1.0.3. What exactly does the generated_with property refer to?

**Comment by msbachler** [comment]
OK. After some diffing I worked out that the most likely issue to why the two .sol.js contract files where so different on the two servers was the ether-pudding library. So I ran npm install -g ether-pudding on the old server (with the new geth) to be sure it had the newest version. (I had run npm undate -g already). I then truffle compiled and truffle deployed, and truffle built. That dapp I was testing with now works in the same way as it does on the new server. They both get some promise warnings, which I will look at later but at least they both work now. So I can go ahead and update the new server to the new Geth without worrying. I am not sure I really understand npm and how it updates stuff very well. I will just keep checking versions on the too for all dependent libraries are as required as npm update is obviously not updating as I expected.
